src/toil/common.py

```diff
@@ -1107,3 +1142,28 @@ def getFileSystemSize(dirPath):
     freeSpace = diskStats.f_frsize * diskStats.f_bavail
     diskSize = diskStats.f_frsize * diskStats.f_blocks
     return freeSpace, diskSize
+
+
+def getNodeID(nodeIDFile, nodeIDCommand):
+    """
+    Gets the unique toil node ID for a machine.
+
+    Assumes that at least one of its inputs is None
+
+    :param str nodeIDFile: absolute path to a file with nodeID
+    :param str nodeIDCommand: a command to be run which returns the nodeID
+    :return: a unique strin
+    :rtype: str
+    """
+    if nodeIDCommand is not None:
+        try:
+            output = subprocess.check_output(nodeIDCommand.split())
+        except subprocess.CalledProcessError as e:
```

Comment:
Because we need to catch the output and save the uuid. check_call only gives you the return value while check_output give the output. I was a little unsure about throwing this exception here, but I could not think of a more elegant way to exit.